src/index.d.ts

```diff
@@ -12,3 +12,39 @@
 export { Request, Response } from '@adobe/helix-fetch';
 export { Resolver } from './resolver';
 export * from './adapter';
+
+import {UniversalContext as _UniversalContext} from './adapter';
+
+/**
+ * Namespace declaration with interfaces that wrappers can extend.
+ * 
+ * @example
+ * ```ts
+ * // Extend in a wrapper declaration via merging
+ * declare module '@adobe/helix-universal' {
+ *   namespace Helix {
+ *     export interface UniversalContext {
+ *       foo: () => void;
+ *     }
+ *   }
+ * }
+ * ```
+ * 
+ * @example
+ * ```ts
+ * // Use merged interface as a type
+ * import type { Helix } from '@adobe/helix-universal';
+ * 
+ * async function main(request: Request, context: Helix.UniversalContext) {
+ *   const bar = context.foo();
+ * }
+ * 
+ * ```
+ */
+declare module '@adobe/helix-universal' {
+  namespace Helix {
+    export interface UniversalContext extends _UniversalContext {
+      [key: string]: unknown;
```

Comment:
what is this?

Reply:
it's an index signature, basically means that anything can add to the context with anything it wants that doesn't already exist without throwing a type error, it just identifies the type as `unknown`

all types can be assigned to `unknown`, but `unknown` needs to be cast to be assigned to a typed variable

without it:
```ts
export function main(req: Request, ctx: Helix.UniversalContext) {
  ctx.data = "foo"; // Error, key `data` doesn't exist on type Helix.UniversalContext
}
```

with it:
```ts
export function main(req: Request, ctx: Helix.UniversalContext) {
  ctx.data = "foo";
  const data = ctx.data; // type = unknown
  const dataTyped: string = ctx.data as string; // type = string
}
```

covers the 2 cases:
1.  strongly typing context data that is added by the service (ie. Admin API) -> create an interface and cast or extend the namespace
2. context is changed and exposed inside `main` by a wrapper -> wrapper type decl modifies the namespace for you

Reply:
thanks for the explanation.